libs/algorithms/include/hpx/parallel/util/detail/handle_local_exceptions.hpp

```diff
@@ -144,46 +164,66 @@ namespace hpx { namespace parallel { namespace util { namespace detail {
         ///////////////////////////////////////////////////////////////////////
         HPX_NORETURN static void call(std::exception_ptr const&)
         {
+#if defined(HPX_COMPUTE_DEVICE_CODE)
+            HPX_ASSERT(false);
+#else
             hpx::terminate();
+#endif
         }
 
         HPX_NORETURN static void call(
             std::exception_ptr const&, std::list<std::exception_ptr>&)
         {
+#if defined(HPX_COMPUTE_DEVICE_CODE)
+            HPX_ASSERT(false);
```

Comment:
We might want to call `std::terminate()` here (is there such thing in device code?).

Reply:
As far as I know, no. I think an assertion is the closest we can get that kind of behaviour (with the downside that it's only enabled in debug mode). This fails fairly gracefully with the messages being printed and the next device synchronization failing (which we turn into an exception).